### PR TITLE
Rework SPDK image build process.

### PR DESCRIPTION
**Pull request by artek-koltun**
* By default SPDK is built with x86-64-v2 to cover as many platforms as possible. Additioanlly OPTIMIZE_SPDK=true env var can be set to build SPDK optimized for current build platform.
* Remove SPDK build scripts and move logic into Dockerfile.

### CODE REVIEW

File: build/storage/scripts/build_container.sh [hunk 1]
```diff
@@ -13,22 +13,30 @@ root_dir="${script_dir}/.."
 source "${script_dir}"/spdk_version.sh
 "${script_dir}"/prepare_to_build.sh
 
+possible_containers=("storage-target" "ipu-storage-container" \
+    "host-target" "traffic-generator" "test-driver" "ipdk-unit-tests" \
+    "cmd-sender")
+container_to_build="${1}"
```
Comment (intelfisz) [inline build/storage/scripts/build_container.sh:19]: Don't we check if the container to build is one from the allow list?

**Review by intelfisz — COMMENTED** [review]
Looks good! Small remarks:
I don't see any documentation mentioning new build option?
Also please correct some typos in the commit message (if it is the same as in the conversation).

File: build/storage/scripts/build_container.sh (hunk 1 above)
Comment (artek-koltun) [reply build/storage/scripts/build_container.sh:19]: Yes, we do check, but in another place on line 50. I will move the definitions back closer to that check

Comment (artek-koltun) [reply build/storage/scripts/build_container.sh]: Yes, we do check, but in another place on line 50. I will move the definitions back closer to that check

**Review by intelfisz — APPROVED** [review]

**Pull request merged**